### ISSUE TITLE
Trigger Agent struggles

### ISSUE THREAD

**Issue by geezfools** [huginn/huginn#3209]
Goal. Scan several websites that sell disc golf discs. Either by RSS or website agent. Once they make it into huginn, those will move to a trigger agent that decides if they move to a post agent (for discord) or get ignored. I'm attempting to sort by the manufacturer. For example, Innova, Discraft, Discmania, etc
All of my agents to monitor the sites are working correctly and were making it to discord, unfiltered.

I created a trigger agent and was having issues, so I went crazy and created this. Yet, I still have issues with items either not making it through or items that DO NOT match those keywords, do make it through

```
{
  "expected_receive_period_in_days": "365",
  "keep_event": "false",
  "rules": [
    {
      "type": "regex",
      "value": "innova|champion|star|DX|Gstar|XT|Pro",
      "path": "text"
    },
    {
      "type": "regex",
      "value": "innova|champion|star|DX|Gstar|XT|Pro",
      "path": "title"
    },
    {
      "type": "regex",
      "value": "innova|champion|star|DX|Gstar|XT|Pro",
      "path": "url"
    },
    {
      "type": "regex",
      "value": "innova|champion|star|DX|Gstar|XT|Pro",
      "path": "description"
    },
    {
      "type": "regex",
      "value": "innova|champion|star|DX|Gstar|XT|Pro",
      "path": "message"
    }
  ],
  "message": "{{title}} - {{urls | first}}"
}
```

**Comment by geezfools** [comment]
One of the events that made it through from an RSS agent

```
{
  "message": "Route Package Protection - https://sabattusdiscgolf.com/products/routeins"
}
```

One of the events that was ignored, I'm assuming...

```
{
  "title": "2023 Innova Calendar",
  "link": "https://discgolfunited.com/2023-innova-disc-golf-calendar.html",
  "url": "https://discgolfunited.com/2023-innova-disc-golf-calendar.html",
  "author": "Disc Golf United"
}
```

**Comment by Unending** [comment]
If those are real and complete examples neither should be passed by the trigger agent.

Consider that `By default, all rules must match for the Agent to trigger`, a missing path should mean a failed match. Also keep in mind that *Route Package Protection - https://sabattusdiscgolf.com/products/routeins* contains the word "products" which would match `Pro` since `regex patterns are matched case insensitively` and also contains the word "Protection" which would match it even case sensitive.

Consider a highly restrictive matching pattern and a large test set before committing.

**Comment by geezfools** [comment]
> If those are real and complete examples neither should be passed by the trigger agent.
> 
> Consider that `By default, all rules must match for the Agent to trigger`, a missing path should mean a failed match. Also keep in mind that _Route Package Protection - https://sabattusdiscgolf.com/products/routeins_ contains the word "products" which would match `Pro` since `regex patterns are matched case insensitively` and also contains the word "Protection" which would match it even case sensitive.
> 
> Try a highly restrictive matching pattern and a large test set before committing.

That's helpful. I misunderstood that all would have to match and assumed as long as ONE matched, I'd be good. I will have to rethink how to filter these correctly. Especially for something like the word Pro as it's a popular plastic used by Innova.

Thank you for the quick response, I will dig into this a little more tonight and see if I can make sense of what I'm attempting to accomplish.

**Comment by geezfools** [comment]
Is there a way to specifically filter the word Pro inside a Trigger Agent like I'm using?

Sorry if it's a stupid question, learning as I go here, heh

**Comment by Unending** [comment]
Try the following from the help text:

- By default, all rules must match for the Agent to trigger. You can switch this so that only one rule must match by setting `must_match` to `1`.
- Note that regex patterns are matched case insensitively. If you want case sensitive matching, prefix your pattern with `(?-i)`.

To just match "Pro" you could add a word boundary like `\bPro\b`.

**Comment by virtadpt** [comment]
@Unending This is where `"must_match": 1` comes in handy.

(Note: You don't have to match on only 1.  You can match on 2, 3.. however many you need.)

**Comment by geezfools** [comment]
Thank you both. I have to admit that while I am very much a n00b when it comes to this stuff, the help article and wiki are anything but easy to navigate and find information. I know that having some base knowledge is more or less required to make this all work, but most things are written above my knowledge for sure, so I absolutely appreciate your help with this and hopefully, down the line when someone is trying to figure this stuff out, it can also help them, lol.

After realizing my mistake with my original trigger agent (multiple rules), I took the time to fine tune my rss agents and website agents, making sure that both were pushing out titles and urls and named as such, that helped immensely. I've got some more work to go to dial this in and I'm still not sure if I'm doing this completely right, but things are mostly going to their respective channels on discord so far, lol
